runtime/js/99_main.js

```diff
@@ -75,60 +75,71 @@ delete Object.prototype.__proto__;
   const onerror = () => {};
 
   function postMessage(data) {
-    const dataJson = JSON.stringify(data);
-    const dataIntArray = encoder.encode(dataJson);
+    const dataIntArray = core.serialize(data);
     opPostMessage(dataIntArray);
   }
 
   let isClosing = false;
-  async function workerMessageRecvCallback(data) {
-    const msgEvent = new MessageEvent("message", {
-      cancelable: false,
-      data,
-    });
-
-    try {
-      if (globalThis["onmessage"]) {
-        const result = globalThis.onmessage(msgEvent);
-        if (result && "then" in result) {
-          await result;
-        }
-      }
-      globalThis.dispatchEvent(msgEvent);
-    } catch (e) {
-      let handled = false;
-
-      const errorEvent = new ErrorEvent("error", {
-        cancelable: true,
-        message: e.message,
-        lineno: e.lineNumber ? e.lineNumber + 1 : undefined,
-        colno: e.columnNumber ? e.columnNumber + 1 : undefined,
-        filename: e.fileName,
-        error: null,
+  async function pollForMessages() {
+    while (!isClosing) {
+      const bufferMsg = await opGetMessage();
+      const data = core.deserialize(new Uint8Array(bufferMsg));
```

Comment:
Isn't call to `new Uint8Array` superfluous here?

Reply:
I thought `Box<[u8]>` is auto serialized to Uint8Array... cc @AaronO

Reply:
Actually just checked - `ZeroCopyBuf` / `serde_v8::MagicBuffer` is auto serialized to a `Uint8Array`. You can `ZeroCopyBuf::from(Box<[u8]>)`.

Reply:
I fixed this in my latest changes, also at the host side this was fixed.